### ISSUE TITLE
How `update_topics` re-computes or re-assigns probabilities?

### ISSUE THREAD

**Issue by schoobani** [MaartenGr/BERTopic#1387]
After getting topics, I need to split them based on internal labels. Doing so, I'll need to use `update_topics` with a custom list of topics.  But `update_topics` function does not update `probabilities_` anymore. Later I'll need a probability assigned to each document in my pipeline.

While documentations [point out](https://github.com/MaartenGr/BERTopic/blob/94064441233564f29b5ac03ff5c38669067ba4e1/bertopic/_bertopic.py#L1356C6-L1356C6) that `update_topics` can get custom topics but it doesn't seem to be coherently changing all the states. Am I missing something here?

**Comment by MaartenGr** [comment]
It is important to know that the probabilities are calculated as an inherent process of the underlying clustering algorithm. With `update_topics`, you are essentially updating the topic representations only and not the underlying clustering algorithm. As a result, whenever you update the model with custom topics through this method, there is no calculation of probabilities with the clustering algorithm. 

> Later I'll need a probability assigned to each document in my pipeline.

Do you need a single probability score or a topic-document probability vector for each document? If it is the former, then you can still use the `probabilities_` since they still accurate the underlying topics and assignments. If it is the latter, then that is not possible since the custom assignment generally does not follow a merging procedure like with `merge_topics`. For example, you might map some documents that were originally in topic 0 to topic 3 whereas some will be mapped to topic 2. This means that there is no clear structure for merging topic-document probabilities.

**Comment by schoobani** [comment]
Thanks for your answer, what do you think is the best practice in assigning custom labels to the model? 
Do you suggest using `update_topics` and then assigning a cosine distance between each doc and the average embeddings of the underlying topic? what about using `fit_transform` method again with `y` classes (topics after splitting them as ground truth classes)

**Comment by MaartenGr** [comment]
> Thanks for your answer, what do you think is the best practice in assigning custom labels to the model?

There are several ways to get the probabilities:

First, you can indeed use the cosine similarity between document and topic embeddings to get an idea of the topic distribution in a document. There are a few others that you can use as mentioned in [`.approximate_distribution`](https://maartengr.github.io/BERTopic/getting_started/distribution/distribution.html)

Second, a trick that you can use is to performant [manual](https://maartengr.github.io/BERTopic/getting_started/manual/manual.html) topic modeling with BERTopic. Here, you can use the labels you created afterwards as the target over which it creates the topic representations. After doing so, you can then apply [`.approximate_distribution`](https://maartengr.github.io/BERTopic/getting_started/distribution/distribution.html). The benefit of this approach is that everything is saved internally and since you seldom need the dimensionality reduction and clustering step after training, it is often okay to skip them. In practice, it would look like this:

```python
from bertopic import BERTopic
from bertopic.cluster import BaseCluster
from bertopic.vectorizers import ClassTfidfTransformer
from bertopic.dimensionality import BaseDimensionalityReduction

# Prepare our empty sub-models and reduce frequent words while we are at it.
empty_dimensionality_model = BaseDimensionalityReduction()
empty_cluster_model = BaseCluster()

# Fit BERTopic without actually performing any clustering
topic_model= BERTopic(
        embedding_model=YOUR_EMBEDDING_MODEL,
        umap_model=empty_dimensionality_model,
        hdbscan_model=empty_cluster_model,
)
topics, probs = topic_model.fit_transform(docs, y=y)
```

I would still advise using an embedding model or passing the embeddings directly as that will create the resulting topic embeddings.

**Comment by schoobani** [comment]
Thank you, this was very helpful indeed.
